apps/api/src/services/exchange-rate-data.service.ts

```diff
@@ -1,27 +1,45 @@
-import { currencyPairs } from '@ghostfolio/common/config';
+import { baseCurrency } from '@ghostfolio/common/config';
 import { DATE_FORMAT, getYesterday } from '@ghostfolio/common/helper';
 import { Injectable } from '@nestjs/common';
-import { Currency, DataSource } from '@prisma/client';
+import { DataSource } from '@prisma/client';
 import { format } from 'date-fns';
-import { isEmpty, isNumber } from 'lodash';
+import { isEmpty, isNumber, uniq } from 'lodash';
 
 import { DataProviderService } from './data-provider/data-provider.service';
 import { IDataGatheringItem } from './interfaces/interfaces';
+import { PrismaService } from './prisma.service';
 
 @Injectable()
 export class ExchangeRateDataService {
+  private currencies: string[] = [];
   private currencyPairs: IDataGatheringItem[] = [];
   private exchangeRates: { [currencyPair: string]: number } = {};
 
-  public constructor(private dataProviderService: DataProviderService) {
+  public constructor(
+    private readonly dataProviderService: DataProviderService,
+    private readonly prismaService: PrismaService
+  ) {
     this.initialize();
   }
 
+  public getCurrencies() {
+    return this.currencies?.length > 0 ? this.currencies : [baseCurrency];
```

Comment:
base currency might not be part of `this.currencies`, or? I just try to understand why the `? :` is required

Reply:
If the instance has no transactions and no users, it could be problematic if there is no currency at all